packages/Webkul/Shop/src/Http/Controllers/ProductsCategoriesProxyController.php

```diff
@@ -44,9 +44,13 @@ public function index(Request $request)
             }
 
             if ($product = $this->productRepository->findBySlug($slugOrPath)) {
-                $customer = auth()->guard('customer')->user();
+                if ($product->visible_individually && $product->url_key) {
+                    $customer = auth()->guard('customer')->user();
 
-                return view($this->_config['product_view'], compact('product', 'customer'));
+                    return view($this->_config['product_view'], compact('product', 'customer'));
+                } else {
+                    session()->flash('warning', trans('shop::app.common.product-individual-view-inactive'));
```

Comment:
Instead of adding flash message, return to 404 page

Reply:
> Instead of adding flash message, return to 404 page

Removed flash message, 404 page already returning.